src/dolphin/phase_link/_core.py

```diff
@@ -300,6 +308,11 @@ def run_cpl(
         strides,
         neighbor_arrays=neighbor_arrays,
     )
+    if baseline_lag:
+        iu = np.triu_indices(C_arrays.shape[0], baseline_lag)
+        il = np.tril_indices(C_arrays.shape[0], -baseline_lag)
+        C_arrays = C_arrays.at[iu].set(0)
+        C_arrays = C_arrays.at[il].set(0)
```

Comment:
Ah, this is actually the wrong indexing... the `C_arrays` shape semantics are (`pixel rows, pixel cols, time1, time2`).
So right now, this is zeroing out all the pixels in the image in a tri-diagonal pattern.

```suggestion
    if baseline_lag:
        iu = np.triu_indices(C_arrays.shape[0], baseline_lag)
        il = np.tril_indices(C_arrays.shape[0], -baseline_lag)
        C_arrays = C_arrays.at[:, :, iu].set(0)
        C_arrays = C_arrays.at[:, :, il].set(0)
```


I saw that by [changing the test](https://github.com/isce-framework/dolphin/blob/main/tests/test_phase_link_core.py#L28-L29) (which I can't suggest here)

```

@pytest.mark.parametrize("baseline_lag", [None, 3])
@pytest.mark.parametrize("use_evd", [False, True])
def test_estimation(C_truth, slc_samples, use_evd, baseline_lag):
```

so let's add this option to the basic phase linking test in a parameter